astropy/time/tests/test_methods.py

```diff
@@ -619,6 +629,92 @@ def test_sort(self, use_mask):
             assert np.all(self.t0.sort(-1)[:, :, 0] == self.t0.min(-1))
             assert np.all(self.t0.sort(-1)[:, :, -1] == self.t0.max(-1))
 
+    @pytest.mark.parametrize('axis', [None, 0, 1, 2, (0, 1)])
+    @pytest.mark.parametrize('where', [True, np.array([True, False, True, True, False])[..., np.newaxis]])
+    @pytest.mark.parametrize('keepdims', [False, True])
+    def test_mean(self, use_mask, axis, where, keepdims):
+        self.create_data(use_mask)
+
+        kwargs = dict(axis=axis, where=where, keepdims=keepdims)
+
+        def is_consistent(time):
+
+            where_expected = where & ~time.mask
+            where_expected = np.broadcast_to(where_expected, time.shape)
+
+            kw = kwargs.copy()
+            kw['where'] = where_expected
+
+            divisor = where_expected.sum(axis=axis, keepdims=keepdims)
+
+            if np.any(divisor == 0):
+                with pytest.raises(ValueError):
+                    time.mean(**kwargs)
+
+            else:
+                time_mean = time.mean(**kwargs)
+                time_expected = Time(
+                    *day_frac(
+                        val1=np.ma.getdata(time.tai.jd1).sum(**kw),
+                        val2=np.ma.getdata(time.tai.jd2).sum(**kw),
+                        divisor=divisor
+                    ),
+                    format='jd',
+                    scale='tai',
+                )
+                time_expected._set_scale(time.scale)
+                assert np.all(time_mean == time_expected)
+
+        is_consistent(self.t0)
+        is_consistent(self.t1)
+
+        axes_location_not_constant = [None, 2]
+        if axis in axes_location_not_constant:
+            with pytest.raises(ValueError):
+                self.t2.mean(**kwargs)
+        else:
+            is_consistent(self.t2)
+
+    def test_mean_precision(self, use_mask):
+
+        scale = 'tai'
+        epsilon = 1 * u.ns
+
+        t0 = Time('2021-07-27T00:00:00', scale=scale)
+        t1 = Time('2022-07-27T00:00:00', scale=scale)
+        t2 = Time('2023-07-27T00:00:00', scale=scale)
+
+        t = Time([t0, t2 + epsilon])
+
+        if use_mask == 'masked':
+            t[0] = np.ma.masked
+            assert t.mean() == (t2 + epsilon)
+
+        else:
+            assert t.mean() == (t1 + epsilon / 2)
+
+    def test_mean_dtype(self, use_mask):
+        self.create_data(use_mask)
+        with pytest.raises(ValueError):
+            self.t0.mean(dtype=int)
+
+    def test_mean_out(self, use_mask):
+        self.create_data(use_mask)
+        with pytest.raises(ValueError):
+            self.t0.mean(out=Time(np.zeros_like(self.t0.jd1), format='jd'))
+
+    def test_mean_leap_second(self, use_mask):
```

Comment:
It was there since the `TestArithmetic` class is parameterized. If I try and remove it, I get the complaint: `In test_mean_leap_second: function uses no argument 'use_mask'`. Is there a workaround for this issue?

Reply:
OK, just let it be!